application/common/models/User.php

```diff
@@ -521,22 +541,81 @@ public function getDisplayName()
         return $this->display_name ?? "$this->first_name $this->last_name";
     }
 
+    /**
+     * Based on provided time, determine whether to present a reminder to add
+     * an MFA option.
+     * @param int $now
+     * @return bool
+     * @throws Exception
+     */
+    public function isTimeToNagToAddMfa(int $now): bool
+    {
+        return MySqlDateTime::isBefore($this->nag_for_mfa_after, $now)
+            && (count($this->getVerifiedMfaOptions()) === 0);
+    }
+
+    /**
+     * Based on provided time, determine whether to present a reminder to add
+     * a recovery method option.
+     * @param int $now
+     * @return bool
+     * @throws Exception
+     */
+    public function isTimeToNagToAddMethod(int $now): bool
+    {
+        return MySqlDateTime::isBefore($this->nag_for_method_after, $now)
+            && (count($this->getVerifiedMethodOptions()) === 0);
+    }
+
     /**
      * Based on current time, determine whether to present a profile review to
      * the user.
+     * @param int $now
+     * @return bool
+     * @throws Exception
      */
-    public function isTimeForReview()
+    public function isTimeForReview(int $now)
     {
-        return MySqlDateTime::isBefore($this->review_profile_after, time());
+        return MySqlDateTime::isBefore($this->review_profile_after, $now);
     }
 
+    /**
+     * Based on current time and presence of MFA and Method options,
+     * determine which "nag" to present to the user.
+     *
+     */
+    public function getNagState()
```

Comment:
see my notes earlier about the possibility of a NagState struct or class or something like that to encapsulate this kind of functionality.

Reply:
I'll take a look. Thanks for the tip.

Reply:
👍